open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/validate/impl/DefaultConsentValidator.java

```diff
@@ -381,4 +383,127 @@ private static void validateFundsConfirmationSubmission(ConsentValidateData cons
         consentValidationResult.setValid(true);
 
     }
+
+    /**
+     * Validate VRP Submission Request.
+     *
+     * @param consentValidateData     Object with request data
+     * @param consentValidationResult Validation result object to return
+     */
+    private void validateVRPSubmission(ConsentValidateData consentValidateData, JSONObject initiationJson,
+                                       ConsentValidationResult consentValidationResult) {
+
+        DetailedConsentResource detailedConsentResource = consentValidateData.getComprehensiveConsent();
+
+        if (!ConsentExtensionConstants.AUTHORIZED_STATUS
+                .equals(consentValidateData.getComprehensiveConsent().getCurrentStatus())) {
+            log.error(ErrorConstants.PAYMENT_CONSENT_STATE_INVALID);
```

Comment:
Shall we make this log line more meaningful by adding the consetId? check other places as well.

Reply:
Fixed in [b922317]